Upload leaderboard-emissions-analysis page (#2578)
huggingface/blog: leaderboard-emissions-analysis.md

```

> - Fine-tunes minimize hallucinations, resulting in more accurate and perhaps energy-efficient responses.

Author: Yeah, it's a good point 

I was describing such cases:
```
[[' Do not use any bullet points. Do not use any quotes. Do not use any numbers. Do not use any citations. Do not use any italics. Do not use any bold text. Do not use any underlined text. Do not use any brackets. Do not use any parentheses. Do not use any colons. Do not use any semicolons. Do not use any dashes. Do not use any hyphens. Do not use any exclamation marks. Do not use any question marks. Do not use any quotation marks. Do not use any apostrophes. Do not use any periods. Do not use any ellipses. Do not use any ampersands. Do not use any asterisks. Do not use any plus signs. Do not use any equal signs. Do not use any backslashes. Do not use any forward slashes. Do not use any underscores. Do not use any pipes. Do not use any curly braces. Do not use any square brackets. Do not use any angle brackets. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. Do not use any parentheses. 
```

I think "inaccurate or anomalous outputs, such as repetitive patterns" will be more "accurate" in this case hehe

**Caveat:** This does not mean that model X emits Y CO₂ at inference in general!
Instead, what this means is that model X emitted Y CO₂ on our very specific inference setup, and you can still learn a lot from that 😀

# General Trends

💡 **Note:** Since we wanted to look at general trends, we kept only the most frequent model architectures, and models for which we had the parameter count. 

We therefore looked at 2,796 models from the recent families Gemma/Gemma2, all generations of Llama, Mistral, Mixtral, as well as Phi/Phi3, Qwen2 and above. We also included older model families such as GPT, GPT-NeoX, and T5.

## **“Official Providers” Models**

Official models come from high-quality trusted model creators, such as research groups or community consortiums (EleutherAI, NousResearch), FAANG (Google, Meta, Alibaba…), startups (MistralAI, 01AI), etc, who have taken the time and compute to create new high-quality models. They represent 339 models.

![official_providers_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/official_providers_models.png)

> - As expected, overall, the bigger the model size, the higher the CO₂ cost. However, the increase in performance is not always proportional, leading to diminishing returns on investment (ROI).

Reviewer: ```suggestion
- As expected, overall, the bigger the model size, the higher the CO₂ cost. However, the increase in performance is not always proportional, leading to diminishing returns.
```

    - Models from AbacusAI, Qwen, and AllenAI, around 70B parameters, achieve an average performance score above 40 across multiple evaluation benchmarks.
    - On the other hand, the lowest-ranked models in the top-right quadrant are older models: `Qwen-1.5-100B` models, with `Mixtral8x22B` showing the weakest performance.

>     - Overall, MoEs seem to have a relatively poor ROI. Although these models aim to reduce computational overhead by activating only a subset of their parameters for a given task, some exhibit higher-than-expected CO₂ emissions due to extremely long inference times.

Reviewer: ```suggestion
    - Overall, MoEs seem to have a relatively poor score-to-emission ratio. Although these models aim to reduce computational overhead by activating only a subset of their parameters for a given task, some exhibit higher-than-expected CO₂ emissions due to extremely long inference times.
```

- Smaller models occupy the lower-cost quadrants, making them appealing for use cases where energy efficiency is paramount. Among these, `Qwen-2.5-14B` and `Phi-3-Medium` models seem to have the best score-to-emission ratio.
- Instruction-tuned models often outperform their bases on the leaderboard. However, certain instruct-tuned models can be exceedingly verbose, which inflates both inference time and energy consumption during our generative evaluations (MATH and IFEval). Some instruct-tuned models exhibit another issue: much lower scores than expected for their cost. This occurs when they overfit specific prompt formats, becoming unable to follow the formats expected on the leaderboard, leading mostly to lower scores on MATH evaluations.

## Community Models

As the community focuses largely on small models, it manages to reach up to 35 in average score (best scores are around 45) for models below 10B parameters, for less than 5kg CO₂! 

![community_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/community_models.png)

> However, interestingly, the trend of CO₂ emissions to model size, even at higher values, is not the same between community models and official releases: community fine-tunes or merges tend to be more CO₂ efficient than the official models they start from!

Reviewer: would it make sense to only compare finetuned models from official providers to finetuned from the community ?

Author: In terms of plot, it can be hard to plot only official fine-tunes vs community fine-tunes as we need to somehow isolate them

## High-Parameter Language Models

First, let’s look at three 70B models, comparing the average CO₂ consumption of the base, its official fine-tunes, and community fine-tunes.

- Overall, for `Qwen2.5` and `Llama3.1`, the base models and user fine-tunes tend to exhibit similar CO₂ emissions, but the official fine-tunes consume twice as much energy.
- Curiously, for `Qwen2`, the base model is significantly more energy-intensive than its fine-tunes.
- The strong performance of community fine-tunes might be attributed to their benchmark-specific adaptations, leading to shorter outputs and reduced energy consumption.
    
    ![70b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/70b_models.png)
    

## Compact Language Models

When we examine 7B+ models in the same way, we observe that there is no consistent trend.

- For `Llama3.1` and `Qwen2.5`, the base models consume twice as much as the fine-tunes.
- For `Qwen2` and `Mistral v0.3`, the consumption of the community fine-tunes is higher.
    

>     ![7b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/7b_models.png)

Reviewer: the scale is very big, would be better to reduce the range of the y axis so that we can more easily compare

- By contrast, the base model `Qwen2-72B` is not optimized to follow instructions. Rather than directly answering, it tends to continue the prompt, producing verbose and sometimes less coherent outputs. This illustrates that IFEval is not ideally suited for evaluating base models without instruction tuning.

For example:
```
Q: Write a 300+ word summary of the wikipedia page "[https://en.wikipedia.org/wiki/Raymond_III,_Count_of_Tripoli](https://en.wikipedia.org/wiki/Raymond_III,_Count_of_Tripoli)". Do not use any commas and highlight at least 3 sections that has titles in markdown format, for example *highlighted section part 1*, *highlighted section part 2*, *highlighted section part 3*.
A: Do not use any bullet points. Do not use any quotes. Do not use any numbers. Do not use any citations. Do not use any italics. Do not use any bold text. Do not use any underlined text. Do not use any brackets. Do not use any parentheses. Do not use any colons. Do not use any semicolons. Do not use any dashes. Do not use any hyphens. Do not use any exclamation marks. Do not use any question marks. Do not use any quotation marks. Do not use any apostrophes. Do not use any periods. <…>
```

> - Alternatively, `Qwen2-72B` can produce a very long answer, as in the example below. Interestingly, in this instance, the model appears to have learned to follow instructions rather than merely continuing the prompt:

Reviewer: it's because we do not use stop token for `ifeval` and the base model does not have EOS tokens so it generates as `max_tokens` every time (to check because i'm not 100% sure), while finetunes just hit EOS token at some point

```
A: <…> This post is written in a conversational tone and includes some personal anecdotes. It also includes a photo and a link to a relevant website. The post is less than 300 words, as required. I hope you enjoy reading it! # Japan # Travel # BlogPost\n```markdown\n# Japan: A Land of Contrasts\n\nI recently returned from a trip to Japan, and I'm still trying to process all the experiences I had. From the bustling streets of Tokyo to the serene gardens of Kyoto, Japan is a land of contrasts.\n\nOne of the highlights of
```

Fine-tuning large language models like `Qwen2-72B` and `Meta-Llama-3.1-8B` improves output coherence and conciseness, reducing computational load and potentially CO₂ emissions. However, for now, exact emission data for specific benchmarks is not available, limiting detailed comparisons. Despite this, it is evident that fine-tuning enhances efficiency, though the reason for emission reductions remains uncertain.

# **Future Questions**

Several open questions remain, for interested individuals in the community to explore!

- What underlying factors contribute to the lower emissions of fine-tuned community models compared to pre-trained models?

>     - Could dataset contamination in evaluations like MATH and IFEval lead to artificially improved efficiency by enabling models to terminate inference earlier?

Reviewer: pretty it's the OES token, for example in math we have a stop token argument (`\n\n`) so base model tend to terminate faster than finetunes (we remove stop token for finetunes). To be verified again but I think this could explain it!

Author: A very interesting observation, I think I need to explore it in details and analyse MATH (hard to verify fast)

- Overall, for `Qwen2.5` and `Llama3.1`, the base models and user fine-tunes tend to exhibit similar CO₂ emissions, but the official fine-tunes consume twice as much energy.
- Curiously, for `Qwen2`, the base model is significantly more energy-intensive than its fine-tunes.
- The strong performance of community fine-tunes might be attributed to their benchmark-specific adaptations, leading to shorter outputs and reduced energy consumption.
    
    ![70b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/70b_models.png)
    

## Compact Language Models

When we examine 7B+ models in the same way, we observe that there is no consistent trend.

- For `Llama3.1` and `Qwen2.5`, the base models consume twice as much as the fine-tunes.
- For `Qwen2` and `Mistral v0.3`, the consumption of the community fine-tunes is higher.
    
    ![7b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/7b_models.png)

## Analyzing Emission Patterns in Qwen2 and Llama Models

> 

Reviewer 2: This section is good, but I would split the example analysis a bit more to make it more readable and more easily see when it starts/stops

Author: A good idea, done!

```
Q: Given the sentence "Two young boys with toy guns and horns." can you ask a question? Please ensure that your response is in English, and in all lowercase letters. No capital letters are allowed.
A: The sentence "Two young boys with toy guns and horns" is a simple statement that describes a scene involving two boys who are carrying toy guns and horns. Based on this sentence, we can ask several questions to gain more information about the situation or the boys themselves. Here are a few examples:\n\n1. Who are the two young boys?\n2. What are the boys doing with the toy guns and horns?\n3. Where are the two young boys?\n4. How old are the two young boys?\n5. Are the toy guns and horns part of a game or activity?\n6. Are the two young boys playing together or separately?\n7. What type of toy guns and horns do the boys have?\n8. Are the two young boys wearing any other toys or accessories?\n9. Is there any adult supervision for the two young boys?\n10. How did the two young boys acquire the toy guns and horns?\n\nThese questions can help us understand the context and details of the situation involving the two young boys with toy guns and horns.
```

> One possible explanation is that, for IFEval, we do not use a stop token, and the base model does not have an end-of-sequence (EOS) token. As a result, it may continue generating tokens until it reaches the maximum limit. By contrast, fine-tuned models typically include an EOS token, allowing them to produce more concise responses and terminate appropriately.

Reviewer 2: @NathanHB wdyt of this paragraph?

Reviewer: I agree with this! However, it needs to be verified, we can simply check the size of outputs of the base models on ifeval

Author: I checked briefly and it looks like this

Author: As far as I see – no, but I need to correct this passage: I see there is a `eos_token_id` set for LLama's base models

Author: UPD: Adding the examples of arguments

- IFEval arguments for base models:
```
{ "gen_args_0": { "arg_0": "Write a letter to a friend in all lowercase letters ask them to go and vote.", "arg_1": { "until": [], "do_sample": false, "temperature": 0, "max_gen_toks": 1280 } } }
```

- IFEval arguments for instruct models:
```
{ "gen_args_0": { "arg_0": "<|begin_of_text|><|start_header_id|>user<|end_header_id|>\n\nWrite a letter to a friend in all lowercase letters ask them to go and vote.<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n", "arg_1": { "until": [], "do_sample": false, "temperature": 0, "max_gen_toks": 1280 } } }
```

```

- The community fine-tune generates long responses as the base model and occasionally truncates or repeats text. We can observe this behavior for `Llama-TI-8B` model for the same `doc_id=14` question. Here is only the end of the answer (the full answer is 5,821 characters long):

```

> A: <…> This post is written in a conversational tone and includes some personal anecdotes. It also includes a photo and a link to a relevant website. The post is less than 300 words, as required. I hope you enjoy reading it! # Japan # Travel # BlogPost\n```markdown\n# Japan: A Land of Contrasts\n\nI recently returned from a trip to Japan, and I'm still trying to process all the experiences I had. From the bustling streets of Tokyo to the serene gardens of Kyoto, Japan is a land of contrasts.\n\nOne of the highlights of

Reviewer 2: You can see here that the model seems to answer further than the initial prompt, adding some sort of comment above the task done instead of just accomplishing the task

---
title: "CO₂ Emissions and Models Performance: Insights from the Open LLM Leaderboard" 
thumbnail: /blog/assets/leaderboard-emissions-analysis/thumbnail.png
authors:
- user: alozowski
- user: SaylorTwift
- user: albertvillanova
- user: clefourrier
---

# CO₂ Emissions and Models Performance: Insights from the Open LLM Leaderboard

> # Introduction
> 

Reviewer 3: ```suggestion
```

Your call, I just think it's natural to find an introduction below the title, no need for a heading in my opinion.

Since June 2024, we have evaluated more than 3,000 models on the Open LLM Leaderboard v2, a worldwide ranking of open language models performance. Even though we’re trying to run evaluations without wasting resources (we use the spare cycles of our cluster, in other words the GPUs which are active but waiting between jobs), this still represents quite a big amount of energy spent for model inference!

In the last year, people have become more and more aware that using large language models (LLMs) to generate text has a significant environmental impact, beyond the already important impact of training. Recent research (see [Towards Greener LLMs](https://arxiv.org/abs/2403.20306) article) highlights the challenges of managing resources efficiently at inference due to dynamic and diverse workloads.

> By integrating these scores into the Open LLM Leaderboard, we aim to provide transparency to users about the carbon impact of various model evaluations and hopefully encourage model creators to balance performance with environmental responsibility. 

Reviewer 3: ```suggestion
By integrating carbon emission estimates into the Open LLM Leaderboard, we aim to provide transparency to users about the carbon impact of various model evaluations and hopefully encourage model creators to balance performance with environmental responsibility.
```

"these scores" sounds a bit off to me, I think we had not mentioned anything yet.

Author: Yeah, it sounds better

We were curious to explore the CO₂ emissions associated with model inference and to identify any emerging trends in the data. Along the way, we discovered some surprising insights alongside a few predictable patterns!

> 💡  **Note:** We’ve created a [Colab notebook](https://colab.research.google.com/drive/1gaXUywH9fB5rINM8Ocg_U26u5voDmLUi?usp=sharing) containing all the data and findings discussed here. This notebook allows you to explore the results, run the analyses yourself, and even adapt the code to investigate further questions

Reviewer 3: Perhaps we can use a GitHub or a Hub repo to host the notebook, and use a colab link to it.

Author: Hmm, I don't know how that works, need to check the idea for the next blog / to update this one + I also inserted the colab notebook into our documentation

Reviewer 3: Sure, no worries. Just for info, it's super easy: a notebook file in the Hub automatically gets an "Open in Colab" button: https://huggingface.co/datasets/ariG23498/quick-notebooks/blob/main/nvidia-logits-processor-zoo.ipynb

> # Computing CO₂ cost

Reviewer 3: ```suggestion
## Computing CO₂ cost
```

The title should be the only first-level heading.

Let's take a closer look at how we calculate the CO₂ emissions generated during model inference.

In our case, we use a straightforward heuristic, because all evaluations are run on the same hardware using the same method *(method: loading the model with Transformers and Accelerate using a combination of pipeline parallelism and data parallelism to use our 8 GPUs per node to the fullest)*. It involves factoring in:

- Evaluation time.
- Energy usage based on the power consumption of our cluster’s hardware.
- Carbon intensity of the electricity source powering our hardware.

A detailed explanation and formula can be found in [our documentation](https://huggingface.co/docs/leaderboards/open_llm_leaderboard/emissions).

**Caveat:** This does not mean that model X emits Y CO₂ at inference in general!
Instead, what this means is that model X emitted Y CO₂ on our very specific inference setup, and you can still learn a lot from that 😀

> # General Trends

Reviewer 3: ```suggestion
## General Trends
```

> 💡 **Note:** Since we wanted to look at general trends, we kept only the most frequent model architectures, and models for which we had the parameter count. 

Reviewer 3: ```suggestion
Since we wanted to look at general trends, we only considered the most frequent model architectures, and models for which we had the parameter count. 
```

> We therefore looked at 2,742 models from the recent families Gemma/Gemma2, all generations of Llama, Mistral, Mixtral, as well as Phi/Phi3, Qwen2 and above. We also included older model families such as GPT, GPT-NeoX, and T5.

Reviewer 3: ```suggestion
We therefore looked at 2,742 models from some recent families: Gemma/Gemma2, all generations of Llama, Mistral, Mixtral, as well as Phi/Phi3, Qwen2 and above. We also included older model families such as GPT, GPT-NeoX, and T5.
```

## "Official Providers" Model

> Official models come from high-quality trusted model creators, such as research groups or community consortiums (EleutherAI, NousResearch), FAANG (Google, Meta, Alibaba…), startups (MistralAI, 01AI), etc, who have taken the time and compute to create new high-quality models. They represent 341 models.

Reviewer 3: ```suggestion
Official models come from high-quality trusted model creators, such as research groups or community consortiums (EleutherAI, NousResearch), FAANG (Google, Meta, Alibaba…), startups (MistralAI, 01.AI), etc, who have taken the time and compute to create new high-quality models. They represent 341 models.
```

> ![official_providers_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/official_providers_models.png)

Reviewer 3: What is the "Average" color, the leaderboard score?

(Nothing to do, but curious if you explored logarithmic scales for visualization. Also, I would have probably put the size on the X axis lol)

Author: Yeah, it's the leaderboard score 

Also, I would have probably put the size on the X axis lol –> I'll note it, idk why but we stick to this y-axis and x-axis layout for this analysis :D

- As expected, overall, the bigger the model size, the higher the CO₂ cost. However, the increase in performance is not always proportional, leading to diminishing returns.

>     - Models from AbacusAI, Qwen, and AllenAI, around 70B parameters, achieve an average performance score above 40 across multiple evaluation benchmarks.

Reviewer 3: I was a bit confused whether "performance score" referred to "emissions behaviour". Maybe use "leaderboard score" instead?

Author: Swapping out "performance score" for "leaderboard score", it's indeed more clear + adding "leaderboard score-to-emission ratio"

    - On the other hand, the lowest-ranked models in the top-right quadrant are older models: `Qwen-1.5-100B` models, with `Mixtral8x22B` showing the weakest performance.
    - Overall, MoEs seem to have a relatively poor score-to-emission ratio. Although these models aim to reduce computational overhead by activating only a subset of their parameters for a given task, some exhibit higher-than-expected CO₂ emissions due to extremely long inference times.
- Smaller models occupy the lower-cost quadrants, making them appealing for use cases where energy efficiency is paramount. Among these, `Qwen-2.5-14B` and `Phi-3-Medium` models seem to have the best score-to-emission ratio.

> - Instruction-tuned models often outperform their bases on the leaderboard. However, certain instruct-tuned models can be exceedingly verbose, which inflates both inference time and energy consumption during our generative evaluations (MATH and IFEval). Some instruct-tuned models exhibit another issue: much lower scores than expected for their cost. This occurs when they overfit specific prompt formats, becoming unable to follow the formats expected on the leaderboard, leading mostly to lower scores on MATH evaluations.

Reviewer 3: A couple of labels in the plot for some of the models mentioned could have been helpful for orientation.

Author: Can't figure out how to do it with plotly fast and how to insert an interactive html there, need to think about it later...

Reviewer 3: I meant static labels in the screenshot, just a couple :) But it's a suggestion for a future post, no need to spend more time on it in my opinion.

> ## Community Models

Reviewer 3: ```suggestion
### Community Models
```

As the community focuses largely on small models, it manages to reach up to 35 in average score (best scores are around 45) for models below 10B parameters, for less than 5kg CO₂! 

![community_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/community_models.png)

However, interestingly, the trend of CO₂ emissions to model size, even at higher values, is not the same between community models and official releases: community fine-tunes or merges tend to be more CO₂ efficient than the official models they start from!

> ![all_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/all_models.png)

Reviewer 3: Sorry for all the nit-picking, but this makes community models not sound "official". How about using "Community release" in the legend of this plot and inverting the false/true labels?

Author: Changed all occurrences to "community releases" 👍🏻

Let’s dive deeper into this finding!

> # Detailed Insights

Reviewer 3: ```suggestion
## Detailed Insights
```

Let’s take a close look at high-parameter and compact (> 7B parameters) base models, focusing on three for each category. We will investigate the emissions for each base model itself, for other official fine-tunes, including the official instruct versions, and community fine-tunes.

> ## High-Parameter Language Models

Reviewer 3: ```suggestion
### High-Parameter Language Models
```

First, let’s look at three 70B models, comparing the average CO₂ consumption of the base, its official fine-tunes, and community fine-tunes.

> - Overall, for `Qwen2.5` and `Llama3.1`, the base models and user fine-tunes tend to exhibit similar CO₂ emissions, but the official fine-tunes consume twice as much energy.

Reviewer 3: 🤯

- Curiously, for `Qwen2`, the base model is significantly more energy-intensive than its fine-tunes.
- The strong performance of community fine-tunes might be attributed to their benchmark-specific adaptations, leading to shorter outputs and reduced energy consumption.
    

>     ![70b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/70b_models.png)

Reviewer 3: (nit: community fine-tunes instead of user fine-tunes)

> ## Compact Language Models

Reviewer 3: ```suggestion
### Compact Language Models
```

When we examine 7B+ models in the same way, we observe that there is no consistent trend.

- For `Llama3.1` and `Qwen2.5`, the base models consume twice as much as the fine-tunes.
- For `Qwen2` and `Mistral v0.3`, the consumption of the community fine-tunes is higher.
    
    ![7b_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/7b_models.png)

> ## Analyzing Emission Patterns in Qwen2 Models

Reviewer 3: ```suggestion
### Analyzing Emission Patterns in Qwen2 Models
```

> A significant disparity between the base model and community fine-tunes in `Qwen2-72B` raises intriguing questions about potential differences in verbosity or text quality. The base model exhibits notably higher CO₂ emissions compared to it fine-tunes, while the community fine-tunes demonstrate much lower emissions. 

Reviewer 3: ```suggestion
A significant disparity between the base model and community fine-tunes in `Qwen2-72B` raises intriguing questions about potential differences in verbosity or text quality. The base model exhibits notably higher CO₂ emissions compared to its fine-tunes, while the community fine-tunes demonstrate much lower emissions. 
```

> ### Model Comparison

Reviewer 3: ```suggestion
#### Model Comparison
```

We can compare these three models using our [Comparator tool](https://huggingface.co/spaces/open-llm-leaderboard/comparator):

- `Qwen/Qwen2-72B` (base model)
- `Qwen/Qwen2-72B-Instruct` (official fine-tune)
- `MaziyarPanahi/calme-2.1-qwen2-72b` (community fine-tune)
    
    ![qwen_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/qwen_models.png)
    

> ### Analysis of Model Behavior

Reviewer 3: ```suggestion
#### Analysis of Model Behavior
```

Unfortunately, CO₂ costs per task are not available, so it is unclear whether certain tasks have a disproportionately large impact. However, we know that generative tasks tend to be more costly because they require inferring over more tokens than multiple-choice tasks. With this in mind, we assessed the generative capabilities of these three models and compared their outputs on the IFEval benchmark to identify any notable differences:

- Fine-tuning reduces instances of inaccurate or anomalous outputs (e.g., repetitive patterns), resulting in more energy-efficient responses.
- Additionally, fine-tuning can significantly reduce CO₂ emissions—particularly in the case of `calme-2.1-qwen2-72b`.
- By contrast, the base model `Qwen2-72B` is not optimized to follow instructions. Rather than directly answering, it tends to continue the prompt, producing verbose and sometimes less coherent outputs. This illustrates that IFEval is not ideally suited for evaluating base models without instruction tuning.

> ### Example 1: Base Model Behavior

Reviewer 3: ```suggestion
#### Example 1: Base Model Behavior
```

Here's an example showing how the base model can produce responses with repetivie patterns:
```
Q: Write a 300+ word summary of the wikipedia page "[https://en.wikipedia.org/wiki/Raymond_III,_Count_of_Tripoli](https://en.wikipedia.org/wiki/Raymond_III,_Count_of_Tripoli)". Do not use any commas and highlight at least 3 sections that has titles in markdown format, for example *highlighted section part 1*, *highlighted section part 2*, *highlighted section part 3*.
A: Do not use any bullet points. Do not use any quotes. Do not use any numbers. Do not use any citations. Do not use any italics. Do not use any bold text. Do not use any underlined text. Do not use any brackets. Do not use any parentheses. Do not use any colons. Do not use any semicolons. Do not use any dashes. Do not use any hyphens. Do not use any exclamation marks. Do not use any question marks. Do not use any quotation marks. Do not use any apostrophes. Do not use any periods. <…>
```

> ### Example 2: Verbose Base Model Response

Reviewer 3: ```suggestion
#### Example 2: Verbose Base Model Response
```

> Alternatively, `Qwen2-72B` can produce a very long answer, as in the example below. Interestingly, in this instance, the model appears to have learned to follow instructions rather than merely continuing the prompt:

Reviewer 3: ```suggestion
In other cases, `Qwen2-72B` can produce a very long answer, as in the example below. Interestingly, in this instance, the model appears to have learned to follow instructions rather than merely continuing the prompt:
```

```

One possible explanation is that, for IFEval, we do not use a stop token, and the base model does not have an end-of-sequence (EOS) token. As a result, it may continue generating tokens until it reaches the maximum limit. By contrast, fine-tuned models typically include an EOS token, allowing them to produce more concise responses and terminate appropriately.

> ### Example 3: Comparing Model Responses

Reviewer 3: ```suggestion
#### Example 3: Comparing Model Responses
```

For contrast, here's how the fine-tuned model responds to the same question:

```
Q: Given the sentence "Two young boys with toy guns and horns." can you ask a question? Please ensure that your response is in English, and in all lowercase letters. No capital letters are allowed.
A: what are the two young boys playing with?
```

> ### Example 4: Detailed Response Comparison

Reviewer 3: ```suggestion
#### Example 4: Detailed Response Comparison
```

```

Comparing both responses illustrates how the fine-tuned `calme-2.1-qwen2-72b` version remains more concise, potentially reducing computational overhead and CO₂ emissions, whereas `Qwen2-72B-Instruct` provides a longer, more elaborated story.

> ## Analyzing Emission Patterns in Llama Models

Reviewer 3: ```suggestion
### Analyzing Emission Patterns in Llama Models
```

We can find the same pattern for the `Meta-Llama-3.1-8B` model, although the difference between the base model and the official fine-tune is not that big. Let’s provide the same analysis.

> ### Model Comparison

Reviewer 3: ```suggestion
#### Model Comparison
```

Here is the list of models to compare:

- `meta-llama/Meta-Llama-3.1-8B` (base model)
- `meta-llama/Meta-Llama-3.1-8B-Instruct` (official fine-tune)
- `fluently-lm/Llama-TI-8B` (community fine-tune)
    
    ![llama_models.png](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/leaderboard-emissions-analysis/llama_models.png)

> ### Example 1: Base Model Verbosity

Reviewer 3: ```suggestion
#### Example 1: Base Model Verbosity
```

The base model consistently produces the longest responses, often verbose and prone to repetitive patterns or unnecessary details. This again demonstrates that base models are not optimized to follow instructions and instead tend to continue the prompt. For instance, see the answer for IFEval `doc_id=1`, which totals 5,475 characters:

```
Q: I am planning a trip to Japan, and I would like thee to write an itinerary for my journey in a Shakespearean style. You are not allowed to use any commas in your response.
A: You are also not allowed to use any words that are more than 5 letters long. You are also not allowed to use any words that are more than 5 letters long. You are also not allowed to use any words that are more than 5 letters long. <…> You are also not allowed to use any words that are more than 5 letters long. You are
```
Additionally, some responses are entirely empty (IFEval `doc_id=2` or `doc_id=4`, for example), indicating potential failures or truncation.

> ### Example 2: Instruct Model Improvements

Reviewer 3: ```suggestion
#### Example 2: Instruct Model Improvements
```

```

> ### Example 3: Community Fine-tune Analysis

Reviewer 3: ```suggestion
#### Example 3: Community Fine-tune Analysis
```
